### PR TITLE
#707 Exposed broker log dirs

### PR DESCRIPTION
**Pull request by tdavletov**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "X" next to an item)

- [ ] No need to
- [ ] Manually(please, describe, when necessary)
- [ ] Unit checks
- [x] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings(e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java [hunk 1]
```diff
@@ -786,6 +790,32 @@ private InternalClusterMetrics calculateClusterMetrics(
...
+        .map(admin -> {
+          List<Integer> brokers = new ArrayList<>(cluster.getBrokers());
+          if (reqBrokers != null && !reqBrokers.isEmpty()) {
+            brokers.retainAll(reqBrokers);
+          }
+          return admin.getAdminClient().describeLogDirs(brokers);
+        })
+        .flatMap(result -> Mono.fromSupplier(() -> {
+          try {
+            return result.all().get();
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java:805]: this one is dirty, you should wrap future into mono, please check util classes

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java (hunk 1 above)
Comment (tdavletov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java:805]: @germanosin I don't a better way. Mono has methods for CompletableFuture but not for Future (KafkaFuture in this case). If you have a better solution please advise. One thing I can do is move this code to util class and make this particular place cleaner.

Comment (germanosin) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java:805]: Sure, please use this method https://github.com/provectus/kafka-ui/blob/master/kafka-ui-api/src/main/java/com/provectus/kafka/ui/util/ClusterUtil.java#L57

Comment (tdavletov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java]: Thank you. Fixed.

**Review by IldarAlmakaev — APPROVED** [review]

**Pull request merged**